contracts/SnappAuction.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.5.0;
+
+import "./SnappBase.sol";
+
+
+contract SnappAuction is SnappBase {
+  
+    uint16 public constant AUCTION_BATCH_SIZE = 1000;
+
+    uint public auctionIndex;
+    mapping (uint => PendingBatch) public auctions;
+
+    event SellOrder(
+        uint auctionId, 
+        uint16 slotIndex,
+        uint16 accountId, 
+        uint8 buyToken, 
+        uint8 sellToken, 
+        uint128 buyAmount,
+        uint128 sellAmount
+    );
+
+    constructor () public {
+        auctions[auctionIndex].creationBlock = block.number;
+    }
+
+    function placeSellOrder(
+        uint8 buyToken,
+        uint8 sellToken,
+        uint128 minBuyAmount,
+        uint128 maxSellAmount
+    ) public onlyRegistered() {
+        // Must have 0 < tokenId < MAX_TOKENS anyway, so may as well ensure registered.
+        require(tokenIdToAddressMap[buyToken] != address(0), "Buy token is not registered");
+        require(tokenIdToAddressMap[sellToken] != address(0), "Sell token is not registered");
+
+        // Could also enforce that buyToken != sellToken, but not technically illegal.
+
+        if (
+            auctions[auctionIndex].size == AUCTION_BATCH_SIZE || 
+            block.number > auctions[auctionIndex].creationBlock + 20
+        ) {
+            if (auctionIndex > 1) {
+                require(auctions[auctionIndex - 2].appliedAccountStateIndex != 0, "Too many pending auctions");
```

Comment:
ubernit: This can be one line
```
require(auctionIndex < 2 || auctions[auctionIndex - 2].appliedAccountStateIndex != 0, "Too many pending auctions");
```

Reply:
I prefer 1 liners.

Reply:
But not gonna change it because it makes the line too wide.

Reply:
You could line-break

Reply:
If you insist.